lib/hyacinth/adapters/external_identifier_adapter/hyacinth_metadata.rb

```diff
@@ -2,16 +2,22 @@
 
 # Following module contains functionality to retrieve metadata
 # from the descriptive_metadata hash from DigitalObject
-class Hyacinth::Adapters::ExternalIdentifierAdapter::Datacite::Metadata
-  attr_reader :source, :descriptive_metadata
+class Hyacinth::Adapters::ExternalIdentifierAdapter::HyacinthMetadata
+  DATETIME_PARSER = Hyacinth::DigitalObject::TypeDef::DateTime.new
+  DEFAULT_DATACITE_PROPERTIES = {
+    title: 'Placeholder Title', creators: 'Placeholder Creator',
+    resource_type_general: 'Text', url: 'https://library.columbia.edu',
+    publisher: 'Placeholder Publisher'
+  }.freeze
+
+  attr_reader :source, :descriptive_metadata, :title
   # parse metadata from Hyacinth Digital Objects Data
   # @param digital_object_data_arg [Hash]
   # @api public
-  def initialize(digital_object_data_arg)
-    # dod is shorthand for digital_object_data
-    @source = HashWithIndifferentAccess.new(digital_object_data_arg).freeze
-    # dfd is shorthand for descriptive_metadata
+  def initialize(digital_object)
+    @source = HashWithIndifferentAccess.new(digital_object.as_json).freeze
```

Comment:
Unless there's a reason I'm missing, it seems like turning the digital object instance into a json structure will just make it harder to work with the data (as seen in the created_at, updated_at, etc. methods, where the dates need to be converted from serialized form).  In the rest of the app, we just work with the object itself, so we should probably do that here too.

It's also less error prone to call methods on the object than it is to pass strings to hash accessors.

Reply:
See earlier comment - this is a stepwise refactor to introduce the digital_object as a param (inline with the practice you mention) rather than an already JSON serialized object. But this should be a quick fix.